libiocage/lib/helpers.py

```diff
@@ -103,6 +103,79 @@ def validate_name(name):
     return bool(validate.fullmatch(name))
 
 
+# data -> bool | default
+def parse_bool(data, default=False):
+    """
+    try to parse booleans from strings
+
+    On success, it returns the parsed boolean
+    on failure it returns the `default`.
+    By default, `default` is `False`.
+
+    >>> parse_bool("YES")
+    True
+    >>> parse_bool("false")
+    False
+    >>> parse_bool("/opt/android-studio/build.txt")
+    False
+
+    Note that "-" gets a special treatment:
+
+    >>> parse_bool("-")
+    None
+
+    The behavior of the default parameter can be used to create a
+    pass-thru function:
+
+    >>> data = "/opt/android-studio/build.txt"
+    >>> parse_bool(data, data)
+    "/opt/android-studio/build.txt"
+    """
+
+    if isinstance(data, bool):
+        return data
+    if isinstance(data, str):
+        if data == "-":
+            return None
+        elif data.lower() in ["yes", "true", "on", "1"]:
```

Comment:
We need to be careful with `1` and `0`, because from user input they may be interpreted as string and resolved to a boolean. I'd say those values from user input do not resolve to boolean values at all.

Reply:
yeah, this is extremely context sensitive, and these functions don't have that context.

Reply:
True, but I can't remember where iocage ever allowed 0/1 as boolean values. Let's just drop those two.

Reply:
here's a full listing of potential candidates:

```
 ~/s/t/libiocage  ╍╍ bools  ag -B1 '"[01]"'
libiocage/lib/JailConfig.py
416-    def _default_children_max(self):
417:        return "0"
419-    def _default_allow_set_hostname(self):
420:        return "1"
422-    def _default_allow_sysvipc(self):
423:        return "0"
425-    def _default_allow_raw_sockets(self):
426:        return "0"
428-    def _default_allow_chflags(self):
429:        return "0"
431-    def _default_allow_mount(self):
432:        return "0"
434-    def _default_allow_mount_devfs(self):
435:        return "0"
437-    def _default_allow_mount_nullfs(self):
438:        return "0"
440-    def _default_allow_mount_procfs(self):
441:        return "0"
443-    def _default_allow_mount_zfs(self):
444:        return "0"
446-    def _default_allow_mount_tmpfs(self):
447:        return "0"
449-    def _default_allow_quotas(self):
450:        return "0"
452-    def _default_allow_socket_af(self):
453:        return "0"
464-    def _default_exec_clean(self):
465:        return "1"
467-    def _default_exec_fib(self):
468:        return "0"
494-    def _default_mount_devfs(self):
495:        return "1"
497-    def _default_mount_fdescfs(self):
498:        return "1"

libiocage/lib/JailConfigFstab.py
103-            options="ro",
104:            dump="0",
105:            passnum="0",
144-                "options"    : "ro",
145:                "dump"       : "0",
146:                "passnum"    : "0",

libiocage/lib/Release.py
314-            "basejail"          : False,
315:            "allow_mount_nullfs": "1",
316-            "release"           : self.name,
317:            "securelevel"       : "0"

libiocage/lib/helpers.py
139-            return None
140:        elif data.lower() in ["yes", "true", "on", "1"]:
141-            return True
142:        elif data.lower() in ["no", "false", "off", "0"]:

libiocage/cli/create.py
51-@click.command(name="create", help="Create a jail.")
52:@click.option("--count", "-c", callback=validate_count, default="1",
 ~/s/t/libiocage  ╍╍ bools  
```

Reply:
So what's our response? Removing `0` and `1` from the values that resolve to boolean?